internal/gatewayapi/filters.go

```diff
@@ -681,6 +681,8 @@ func (t *Translator) processExtensionRefHTTPFilter(filter v1beta1.HTTPRouteFilte
 									SafeRegex: header.Value,
 								}
 								rules[i].HeaderMatches = append(rules[i].HeaderMatches, m)
+							case *header.Type == egv1a1.HeaderMatchDistinct && header.Value == nil:
```

Comment:
thanks for catching this, the bug is an unfortunate typo :( 
```
*header.Type == egv1a1.HeaderMatchRegularExpression && header.Value == nil
```
should be changed to 
```
*header.Type == egv1a1.HeaderMatchDistinct && header.Value == nil
```
can you also enhance https://github.com/envoyproxy/gateway/blob/main/internal/gatewayapi/testdata/httproute-with-valid-ratelimitfilter.in.yaml#L52 by adding another `rule` for distinct

Reply:
updated